sockeye/encoder.py

```diff
@@ -331,21 +330,28 @@ def encode(self,
         :return: Encoded versions of input data (data, data_length, seq_len).
         """
 
-        embedding = mx.sym.Embedding(data=data,
-                                     input_dim=self.config.vocab_size,
-                                     weight=self.embed_weight,
-                                     output_dim=self.config.num_embed,
-                                     name=self.prefix + "embed")
-
-        if self.config.num_factors > 0:
-            factor_embeddings = []
-            for i in range(self.config.num_factors):
-                factor_embeddings.append(mx.sym.Embedding(data=factors[i],
-                                                          input_dim=self.config.factor_configs[i].vocab_size,
-                                                          weight=self.embed_factor_weights[i],
-                                                          output_dim=self.config.factor_configs[i].dim,
-                                                          name=self.prefix + "factor" + str(i) + "_embed"))
-            embedding = mx.sym.concat(embedding, *factor_embeddings, dim=2, name=self.prefix + "_concat_embed")
+        if self.config.num_factors == 0:
+            embedding = mx.sym.Embedding(data=data,
+                                         input_dim=self.config.vocab_size,
+                                         weight=self.embed_weight,
+                                         output_dim=self.config.num_embed,
+                                         name=self.prefix + "embed")
+        else:
+            splits = data.split(num_outputs=self.config.num_factors+1, axis=2, squeeze_axis=True)
```

Comment:
not really important, but it would be useful to name this operation in case someone needs to debug. maybe `..., name="%s_split_factors" % self.prefix)`